lib/pliny/librato/metrics/backend.rb

```diff
@@ -7,35 +7,72 @@ module Metrics
       # Implements the Pliny::Metrics.backends API. Puts any metrics sent
       # from Pliny::Metrics onto a queue that gets submitted in batches.
       class Backend
-        attr_reader :queue
+        POISON_PILL = :'❨╯°□°❩╯︵┻━┻'
 
-        def initialize(source: nil, interval: 60, count: 1000, queue: nil)
-          @queue = queue || ::Librato::Metrics::Queue.new(
-            source:              source,
-            autosubmit_interval: interval,
-            autosubmit_count:    count
-          )
-          flush_on_shutdown
+        def initialize(source: nil, interval: 10, count: 500, **opts)
+          @metrics_queue = opts.fetch(:metrics_queue, Queue.new)
+          @librato_queue = opts.fetch(:librato_queue,
+                                      ::Librato::Metrics::Queue.new(
+                                        source:              source,
+                                        autosubmit_interval: interval,
+                                        autosubmit_count:    count
+                                      ))
         end
 
         def report_counts(counts)
-          report(counts)
+          metrics_queue.push(counts)
         end
 
         def report_measures(measures)
-          report(measures)
+          metrics_queue.push(measures)
+        end
+
+        def start
+          start_thread
+          Signal.trap('TERM') { stop }
+          self
+        end
+
+        def stop
+          metrics_queue.push(POISON_PILL)
+          thread.join
         end
 
         private
 
-        def report(metrics)
-          queue.add(metrics)
-        rescue => error
-          Pliny::ErrorReporters.notify(error)
+        attr_reader :librato_queue, :metrics_queue, :thread
+
+        def start_thread
+          @thread = Thread.new 'pliny-librato-metrics-processor' do
```

Comment:
I think the arguments to `Thread.new` are just passed on to the given block. It looks like this is an attempt to give the thread an identifier but I don't think that works?